### PR TITLE
[Transaction] Txn marker channel handle optimize

### PR DESCRIPTION
**Pull request by gaoran10**
# Motivation

End transaction asynchronously.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <log4j2.version>2.13.3</log4j2.version>
     <lombok.version>1.18.4</lombok.version>
     <mockito.version>2.22.0</mockito.version>
-    <pulsar.version>2.8.0-rc-202102252222</pulsar.version>
+    <pulsar.version>2.8.0-rc-202103202224</pulsar.version>
```
Comment (BewareMyPower) [inline pom.xml:50]: Could you explain if the pulsar upgrade is necessary?

Comment (gaoran10) [reply pom.xml:50]: Oh, this is not necessary, I didn't acquire the version `2.8.0-rc-202102252222` at first, maybe network problem. I'll recover the version.

**Comment by codelipenghui** [comment]
https://github.com/streamnative/kop/pull/411 fixes this issue.

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java [hunk 2]
```diff
@@ -73,6 +73,8 @@
 
     private StatsLogger rootStatsLogger;
     private PrometheusMetricsProvider statsProvider;
+    private KoPBrokerLookupManager koPBrokerLookupManager;
```
Comment (BewareMyPower) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java:76]: It's better to change the name to `KopBrokerLookupManager` and `kopBrokerLookupManager` to keep the name consistency with other classes.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerRequestCompletionHandler.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                            txnMetadata.removePartition(topicPartition);
+                            break;
+                        case CORRUPT_MESSAGE:
+                        case MESSAGE_TOO_LARGE:
+                        case RECORD_LIST_TOO_LARGE:
+                        case INVALID_REQUIRED_ACKS: // these are all unexpected and fatal errors
+                            throw new IllegalStateException("Received fatal error " + error.exceptionName()
+                                    + " while sending txn marker for " + transactionalId);
+                        case UNKNOWN_TOPIC_OR_PARTITION:
+//                            case NOT_LEADER_OR_FOLLOWER:
```
Comment (BewareMyPower) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerRequestCompletionHandler.java:149]: remove this line

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
Overall LGTM, just leave two comments

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java (hunk 2 above)
Comment (gaoran10) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java:76]: Ok

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerRequestCompletionHandler.java (hunk 3 above)
Comment (gaoran10) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerRequestCompletionHandler.java]: This condition was introduced in a newer Kafka client version, maybe we could retain this code and recover this code after bumping the Kafka client version, I added a comment.

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**